### PR TITLE
Update to compatible docker compose usage

### PR DESCRIPTION
**Pull request by rolljee**
## What does this PR do?

This PR update kourou to use docker compose instead of docker-compose

### How should this be manually tested?

  - Step 1 : ./bin/run app:doctor
  - Step 2 : ./bin/run instance:kill
  - Step 3 :  ./bin/run instance:spawn
  ...

### CODE REVIEW

**Review by ScreamZ — APPROVED** [review]

File: src/commands/app/doctor.ts [hunk 1]
```diff
@@ -217,16 +217,18 @@ export default class AppDoctor extends Kommand {
...
       if (matches === null || matches.length === 0) {
         this.logKo("Docker Version cannot be found");
       } else {
         this.logOk(`Docker Compose Version: ${matches[1]}`);
       }
     } catch (error: any) {
+      console.log(error);
       this.logKo("Docker Compose cannot be found");
-      suggestions.push("Install Docker Compose with 'npm run install:docker'");
+      suggestions.push("Install docker, It ship docker compose");
```
Comment (Kuruyia) [inline src/commands/app/doctor.ts:231]: I assume that this is a list of suggestions to display to the user?

```suggestion
      suggestions.push("Install Docker and the Compose plugin");
```

File: src/commands/app/start-services.ts [hunk 2]
```diff
@@ -64,22 +64,22 @@ export default class AppStartServices extends Kommand {
     }
 
     this.log(
-      chalk.grey(`\nWriting docker-compose file to ${docoFilename}...\n`)
+      chalk.grey(`\nWriting docker compose file to ${docoFilename}...\n`)
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:67]: Just to add proper capitalization. (Docker Compose is the name of a product)

```suggestion
      chalk.grey(`\nWriting the Docker Compose file to ${docoFilename}...\n`)
```

File: src/commands/app/start-services.ts [hunk 3]
```diff
@@ -97,35 +97,35 @@ export default class AppStartServices extends Kommand {
...
+        title: `docker compose exists and the version is at least ${MIN_DOCO_VERSION}`,
         task: async () => {
           try {
-            const docov = await execute("docker-compose", "-v");
+            const docov = await execute("docker", "compose", "version");
             const matches = docov.stdout.match(/[^0-9.]*([0-9.]*).*/);
             if (matches === null) {
               throw new Error(
-                "Unable to read docker-compose verson. This is weird."
+                "Unable to read docker compose verson. This is weird."
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:107]: Also, maybe replace the "This is weird" with something else? Doesn't sound professional.

```suggestion
                "Unable to read the version of Docker Compose. This is weird."
```

File: src/commands/app/start-services.ts [hunk 4]
```diff
@@ -97,35 +97,35 @@ export default class AppStartServices extends Kommand {
...
               );
             }
             const docoVersion = matches.length > 0 ? matches[1] : null;
 
             if (docoVersion === null) {
               throw new Error(
-                "Unable to read docker-compose version. This is weird."
+                "Unable to read docker compose version. This is weird."
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:114]: Same comment as above. Also, seeing this string duplicated tells me that the check could be refactored.

```suggestion
                "Unable to read the version of Docker Compose. This is weird."
```

File: src/commands/app/start-services.ts [hunk 5]
```diff
@@ -97,35 +97,35 @@ export default class AppStartServices extends Kommand {
...
-            const docov = await execute("docker-compose", "-v");
+            const docov = await execute("docker", "compose", "version");
             const matches = docov.stdout.match(/[^0-9.]*([0-9.]*).*/);
             if (matches === null) {
               throw new Error(
-                "Unable to read docker-compose verson. This is weird."
+                "Unable to read docker compose verson. This is weird."
               );
             }
             const docoVersion = matches.length > 0 ? matches[1] : null;
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:110]: This check seems wrong. It should check that there is at least two matches: `matches.length >= 2`

File: src/commands/app/start-services.ts [hunk 6]
```diff
@@ -97,35 +97,35 @@ export default class AppStartServices extends Kommand {
...
               throw new Error(
-                "Unable to read docker-compose version. This is weird."
+                "Unable to read docker compose version. This is weird."
               );
             }
             try {
               if (docoVersion < MIN_DOCO_VERSION) {
                 throw new Error(
-                  `The detected version of docker-compose (${docoVersion}) is not recent enough (${MIN_DOCO_VERSION})`
+                  `The detected version of docker compose (${docoVersion}) is not recent enough (${MIN_DOCO_VERSION})`
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:120]:
```suggestion
                  `Your version of Docker Compose (${docoVersion}) is below the required version (${MIN_DOCO_VERSION}).`
```

File: src/commands/app/start-services.ts [hunk 7]
```diff
@@ -97,35 +97,35 @@ export default class AppStartServices extends Kommand {
...
                 );
               }
             } catch (error: any) {
               throw new Error(error);
             }
           } catch (error: any) {
             throw new Error(
-              "No docker-compose found. Are you sure docker-compose is installed?"
+              "No docker compose found. Are you sure docker is installed?"
```
Comment (Kuruyia) [inline src/commands/app/start-services.ts:128]:
```suggestion
              "Docker Compose couldn't be found. Are you sure Docker and the Compose plugin are installed?"
```

File: src/commands/instance/spawn.ts [hunk 8]
```diff
@@ -127,23 +127,25 @@ export default class InstanceSpawn extends Kommand {
       );
     }
 
-    this.log(chalk.grey(`\nWriting docker-compose file to ${docoFilename}...`));
+    this.log(chalk.grey(`\nWriting docker compose file to ${docoFilename}...`));
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:130]:
```suggestion
    this.log(chalk.grey(`\nWriting the Docker Compose file to ${docoFilename}...`));
```

File: src/commands/instance/spawn.ts [hunk 9]
```diff
@@ -183,18 +184,18 @@ export default class InstanceSpawn extends Kommand {
...
           chalk.grey("If you want to investigate the problem, try running")
         );
         this.log(
           chalk.grey(
-            `  docker-compose -f ${docoFilename} -p stack-${portIndex} up\n`
+            `  docker compose -f ${docoFilename} -p stack-${portIndex} up\n`
           )
         );
-        throw new Error("docker-compose exited witn non-zero status");
+        throw new Error("docker compose exited witn non-zero status");
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:198]: I don't really know about capitalization here. If this text is printed directly to the user, having a lowercase first letter would be strange.

```suggestion
        throw new Error("docker compose exited with a non-zero status");
```

File: src/commands/instance/spawn.ts [hunk 10]
```diff
@@ -203,35 +204,36 @@ export default class InstanceSpawn extends Kommand {
...
         task: async () => {
           try {
-            const docov = await execute("docker-compose", "-v");
+            const docov = await execute("docker", "compose version");
             const matches = docov.stdout.match(/[^0-9.]*([0-9.]*).*/);
+
             if (matches === null) {
               throw new Error(
-                "Unable to read docker-compose verson. This is weird."
+                "Unable to read docker compose verson. This is weird."
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:215]:
```suggestion
                "Unable to read the version of Docker Compose. This is weird."
```

File: src/commands/instance/spawn.ts [hunk 11]
```diff
@@ -203,35 +204,36 @@ export default class InstanceSpawn extends Kommand {
...
               );
             }
             const docoVersion = matches.length > 0 ? matches[1] : null;
 
             if (docoVersion === null) {
               throw new Error(
-                "Unable to read docker-compose version. This is weird."
+                "Unable to read docker compose version. This is weird."
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:222]: Same comment as above about refactoring.

```suggestion
                "Unable to read the version of Docker Compose. This is weird."
```

File: src/commands/instance/spawn.ts [hunk 12]
```diff
@@ -203,35 +204,36 @@ export default class InstanceSpawn extends Kommand {
...
               );
             }
             try {
               if (docoVersion < MIN_DOCO_VERSION) {
                 throw new Error(
-                  `The detected version of docker-compose (${docoVersion}) is not recent enough (${MIN_DOCO_VERSION})`
+                  `The detected version of docker compose (${docoVersion}) is not recent enough (${MIN_DOCO_VERSION})`
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:228]:
```suggestion
                  `Your version of Docker Compose (${docoVersion}) is below the required version (${MIN_DOCO_VERSION}).`
```

File: src/commands/instance/spawn.ts [hunk 13]
```diff
@@ -203,35 +204,36 @@ export default class InstanceSpawn extends Kommand {
...
                 );
               }
             } catch (error: any) {
               throw new Error(error);
             }
           } catch (error: any) {
             throw new Error(
-              "No docker-compose found. Are you sure docker-compose is installed?"
+              "No docker compose found. Are you sure docker compose is installed?"
```
Comment (Kuruyia) [inline src/commands/instance/spawn.ts:236]:
```suggestion
              "Docker Compose couldn't be found. Are you sure Docker and the Compose plugin are installed?"
```

File: src/commands/app/start-services.ts (hunk 3 above)
Comment (rolljee) [reply src/commands/app/start-services.ts:107]: Well, I'll refactor all of this function and use it inside those three files, I don't like the fact that this has been duplicated everywhere

File: src/commands/app/doctor.ts [hunk 14]
```diff
@@ -216,17 +216,10 @@ export default class AppDoctor extends Kommand {
...
-    } catch (error: any) {
-      this.logKo("Docker Compose cannot be found");
-      suggestions.push("Install Docker Compose with 'npm run install:docker'");
+    const result = await checkPrerequisites(this);
+
+    if (!result) {
+      suggestions.push("Install Docker and the Compose plugin");
     }
 
     this.log(`----------------- DoKtor finish his job ! -----------------`);
```
Comment (Kuruyia) [inline src/commands/app/doctor.ts:225]: One last thing :sweat_smile: 

```suggestion
    this.log(`----------------- DoKtor finished its job ! -----------------`);
```

**Review by Kuruyia — CHANGES_REQUESTED** [review]

**Review by Kuruyia — APPROVED** [review]
LGTM!

**Pull request merged**